5.3b_applying_halo_mass_function.py

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python
+
+from halo_mass_function import HaloMassFunction,Overdensities
+import numpy as np
+import matplotlib.pyplot as plt
+
+save_fig=True #save the figure or not?
+show_fig=False #show the figure or not?
+fontsize=15
+
+#Units and constants
+M_array = np.linspace(1e12,1e20,1000)
+dM = (1e20-1e12)/1000
+z=0
+
+#Calculation of number of halos
+N_array = HaloMassFunction(redshift=z).dndm(mass=M_array)#*M_array
+N_total = np.sum(N_array*dM)
```

Comment:
This is not very accurate because the integration is done in linear rather than log space. Also, consider using a standard library integration routine like np.trapz instead of hand-rolling your own integrations